roles/nova-data/tasks/docker.yml

```diff
@@ -10,11 +8,32 @@
     dest={{ nova.driver.docker.dest }}
     version={{ nova.driver.docker.rev }}
     update={{ openstack.git_update }}
+  register: git_novadocker
+  until: git_novadocker|success
+  retries: 3
+  delay: 60
+
+- name: pip install novadocker
+  pip: name='{{ nova.driver.docker.dest }}'
+       extra_args='-i {{ openstack.pypi_mirror }}'
+  register: result
+  until: result|success
+  retries: 3
+  delay: 60
+  when: git_novadocker|success and openstack_install_method == 'source'
```

Comment:
If git_novadocker is not successful, ansible will error the host and the pip task(s) would never be reached. I think the first part of these when statements are unnecessary.